infra-rpi/__main__.py

```diff
@@ -21,24 +21,38 @@
     )
 
 # Declare command to install k3s on the remote node
-install_cmd = Output.concat(
-    "curl -sfL https://get.k3s.io | sh -s - server --tls-san ", ip_address, " --write-kubeconfig-mode 644 --no-deploy traefik --no-deploy servicelb --data-dir /mnt/data/k3s"
+installk3s_cmd = Output.concat(
+    "curl -sfL https://get.k3s.io | sh -s - server --tls-san ", server_ip_address, " --write-kubeconfig-mode 644 --no-deploy traefik --no-deploy servicelb --data-dir /mnt/data/k3s"
     )
 
-# Send the k3s installation command to the remote note
+# Register k3s install on create and uninstall on delete with the remote node
 install_k3s = command.remote.Command(
-    'asterion-infra-rpi-install-k3s',
+    'asterion-infra-rpi-k3s-installer',
     connection=connection,
-    create=install_cmd.apply(lambda v: v)
+    create=installk3s_cmd.apply(lambda v: v),
+    delete="/usr/local/bin/k3s-uninstall.sh"
 )
 
-# Declare command to retrieve kubeconfig from the remote node
-retrieve_kubeconfig = command.remote.Command('asterion-infra-retrieve-kubeconfig',
+# Implement retrieval of kubeconfig from the remote node
+output_kubeconfig = command.remote.Command(
+    'asterion-infra-rpi-retrieve-kubeconfig',
     connection=connection,
     create='cat /etc/rancher/k3s/k3s.yaml',
     opts=pulumi.ResourceOptions(depends_on=[install_k3s])
 )
 
+# Declare command to set kubeconfig file on the local environment
+setkubeconfig_cmd = Output.concat(
+    "echo '", output_kubeconfig.stdout, "' > ~/.kube/config && sed -i 's/127.0.0.1/", server_ip_address, "/g' ~/.kube/config"
```

Comment:
I would prefer that kubeconfig files stay within the project directory rather than overwriting what a user might already have as their primary kubeconfig file in their home directory.  You can use the same paths we used to, i.e. `infra-rpi/asterion-infra-kubeconfig`.

Reply:
Thanks @jmhbnz - I've pushed up an alternative method using environment variables. I've applied the env vars for all users, and needed to run sudo commands locally to ensure these vars persist. I've also updated the cleanup code to change these vars back to default and remove any added configuration files.

Reply:
Thanks for the updates, we're getting close! Just some feedback around the approach for user configurable kubepath below 🙏🏻